plugins/module_utils/network/ios/facts/vlans/vlans.py

```diff
@@ -44,161 +45,67 @@ def __init__(self, module, subspec="config", options="options"):
 
         self.generated_spec = utils.generate_dict(facts_argument_spec)
 
-    def get_vlans_data(self, connection, configuration):
+    def get_vlans_data(self, connection):
         """Checks device is L2/L3 and returns
         facts gracefully. Does not fail module.
         """
-        if configuration:
-            cmd = "show running-config | sec ^vlan configuration .+"
-        else:
-            cmd = "show vlan"
         check_os_type = connection.get_device_info()
         if check_os_type.get("network_os_type") == "L3":
             return ""
-        return connection.get(cmd)
+        return connection.get("show vlan")
+
+    def get_vlan_conf_data(self, connection):
+        return connection.get("show running-config | sec ^vlan configuration .+")
```

Comment:
Maybe? I don't know if the section filter would work without quotes.
```suggestion
        return connection.get("show running-config | section '^vlan configuration .+'")
```